src/czar/HttpCzarIngestModuleBase.cc

```diff
@@ -0,0 +1,374 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "czar/HttpCzarIngestModuleBase.h"
+
+// System headers
+#include <algorithm>
+#include <stdexcept>
+#include <vector>
+#include <unordered_map>
+
+// Third party headers
+#include "boost/algorithm/string.hpp"
+
+// Qserv headers
+#include "cconfig/CzarConfig.h"
+#include "http/AsyncReq.h"
+#include "http/BinaryEncoding.h"
+#include "http/Client.h"
+#include "http/ClientConfig.h"
+#include "http/ClientConnPool.h"
+#include "http/Exceptions.h"
+#include "http/MetaModule.h"
+#include "http/RequestBodyJSON.h"
+#include "qhttp/Status.h"
+
+using namespace std;
+namespace asio = boost::asio;
+namespace cconfig = lsst::qserv::cconfig;
+namespace http = lsst::qserv::http;
+using json = nlohmann::json;
+
+namespace {
+
+size_t countDirectors(const json& database) {
+    size_t numDirectors = 0;
+    for (const auto& table : database.at("tables")) {
+        if (table.at("is_director").get<int>() != 0) {
+            ++numDirectors;
+        }
+    }
+    return numDirectors;
+}
+
+// These parameters correspond to the present partitioning model of 150k chunks.
+// In reality the parameters are not needed for ingesting regular tables which is
+// is the only kind of tables supported by the current implementation of the module.
+// Some value of the parameters are still required by the Replication/Ingest system's API.
+unsigned int defaultNumStripes = 340;
+unsigned int defaultNumSubStripes = 3;
+float defaultOverlap = 0.01667;
```

Comment:
These appear to be placeholder values that aren't really used? It would be nice to get rid of them to prevent future confusion.

Reply:
Empty (zero) values would work as well. This is related to your next comment.